File: 0179-largest-number/0179-largest-number.cpp
#include <vector>
#include <string>
#include <algorithm>

class Solution {
public:
    std::string largestNumber(std::vector<int>& nums) {
        std::vector<std::string> strNums;
        for (int num : nums) {
            strNums.push_back(std::to_string(num));
        }
        
        std::sort(strNums.begin(), strNums.end(), [](const std::string& a, const std::string& b) {
            return b + a < a + b;
        });
        
        if (strNums[0] == "0") {
            return "0";
        }
        
        std::string result;
        for (const std::string& str : strNums) {
            result += str;
        }
        
        return result;
    }
};

File: 0179-largest-number/0179-largest-number.java
class Solution {
    public String largestNumber(int[] nums) {
        
        String[] strNums = new String[nums.length];
        for (int i = 0; i < nums.length; i++) {
            strNums[i] = String.valueOf(nums[i]);
        }
        
        Arrays.sort(strNums, new Comparator<String>() {
            public int compare(String a, String b) {
                String ab = a + b;
                String ba = b + a;
                return ba.compareTo(ab); // Reverse order
            }
        });
        
        if (strNums[0].equals("0")) {
            return "0";
        }
        
        StringBuilder result = new StringBuilder();
        for (String str : strNums) {
            result.append(str);
        }
        return result.toString();
    }
}
